src/components/cc-product-list/products-controller.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @typedef {import('./cc-product-list.types.js').CategoryData} CategoryData
+ * @typedef {import('./cc-product-list').CcProductList} CcProductList
+ * @typedef {import('./cc-product-list.types.js').Product} Product
+ * @typedef {import('./cc-product-list.types.js').CategoryFilter} CategoryFilter
+ */
+
+export class ProductsController {
+
+  /**
+   * @param {CcProductList} host
+   */
+  constructor (host) {
+    /** @type {CcProductList} The host. */
+    this._host = host;
+
+    /** @type {CategoryData[]} The initial category data list. */
+    this._categoryDataList = [];
+
+    /** @type {CategoryData[]} category data list filtered by the category selected nor with the input */
+    this._currentCategoryDataList = [];
+
+    /** @type {CategoryFilter[]}  */
+    this._categoriesFilters = [];
+
+    /** @type {'all'|string}  */
+    this._currentCategoryNameFilter = 'all';
+  }
+
+  getCategoryDataList () {
+    return this._currentCategoryDataList;
+  }
+
+  /**
+   * @param {CategoryData[]} categoryDataList
+   */
+  set categoryDataList (categoryDataList) {
+
+    this._categoryDataList = categoryDataList;
+    this._currentCategoryDataList = this._categoryDataList;
+    this._categoriesFilters = this._categoryDataList.map(({ categoryName }) => ({ categoryName, toggled: false }));
+    this._currentCategoryNameFilter = 'all';
+  }
+
+  getCategories () {
+    return this._categoriesFilters;
+  }
+
+  /**
+   * @param {string} categoryName
+   */
+  filterCategory (categoryName) {
```

Comment:
nitpick: it's not easy to see that this is a toggle. Maybe think about renaming the method? Or add some comments?